indico/modules/events/timetable/forms.py

```diff
@@ -52,7 +52,7 @@ def __init__(self, *args, **kwargs):
         self.event = kwargs['event']
         self.session_block = kwargs.get('session_block')
         self.day = kwargs.pop('day')
-        if self._default_duration is not None:
+        if self._default_duration is not None and 'time' not in kwargs:
```

Comment:
do you really want to skip the default-duration logic too here? otherwise i'd simply change the next line to use `setdefault`